lib/manager/pub/artifacts.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { exec as _exec } from 'child_process';
+import _fs from 'fs-extra';
+import { join } from 'upath';
+import { envMock, mockExecAll } from '../../../test/exec-util';
+import { mocked } from '../../../test/util';
+import { setAdminConfig } from '../../config/admin';
+import type { RepoAdminConfig } from '../../config/types';
+import * as docker from '../../util/exec/docker';
+import * as _env from '../../util/exec/env';
+import type { UpdateArtifactsConfig } from '../types';
+import * as pub from './artifacts';
+
+jest.mock('fs-extra');
+jest.mock('child_process');
+jest.mock('../../util/exec/env');
+jest.mock('../../util/git');
+jest.mock('../../util/http');
+
+const fs: jest.Mocked<typeof _fs> = _fs as any;
+const exec: jest.Mock<typeof _exec> = _exec as any;
+const env = mocked(_env);
+
+const adminConfig: RepoAdminConfig = {
+  localDir: join('/tmp/github/some/repo'), // `join` fixes Windows CI
+};
+
+const config: UpdateArtifactsConfig = {};
+
+describe('.updateArtifacts()', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.resetModules();
+
+    env.getChildProcessEnv.mockReturnValue(envMock.basic);
+    setAdminConfig(adminConfig);
+    docker.resetPrefetchedImages();
+  });
+  afterEach(() => {
+    setAdminConfig();
+  });
+  it('returns null if no pubspec.lock found', async () => {
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await pub.updateArtifacts({
+        packageFileName: 'pubspec.yaml',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+  it('returns null if updatedDeps is empty', async () => {
+    expect(
+      await pub.updateArtifacts({
+        packageFileName: 'pubspec.yaml',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+  it('returns null if unchanged', async () => {
+    const execSnapshots = mockExecAll(exec);
+    fs.readFile.mockResolvedValueOnce('Current pubspec.lock' as any);
+    fs.readFile.mockResolvedValueOnce('Current pubspec.lock' as any);
```

Comment:
```suggestion
    fs.readFile.mockResolvedValueOnce('Current pubspec.lock');
    fs.readFile.mockResolvedValueOnce('Current pubspec.lock');
```

remove all those `as any`  shouldn't required.

Reply:
Those `any`s are used everywhere in other tests and if they are not used it doesn't work because the function expects a `Buffer`.